src/modules/commander/Commander.cpp

```diff
@@ -3975,6 +3992,8 @@ void Commander::data_link_check(bool &status_changed)
 
 			if (_datalink_last_status_avoidance_system == telemetry_status_s::MAV_STATE_ACTIVE) {
 				mavlink_log_info(&mavlink_log_pub, "Avoidance system healthy");
+				status_flags.avoidance_system_valid = true;
+				avoidance_waiting_count = 0;
```

Comment:
You don't reset `avoidance_waiting_count` here, so you never expect to get it back?